### ISSUE TITLE
When using kustomize, namePrefix should not be required

### ISSUE THREAD

**Issue by denniswebb** [argoproj/argo-cd#1520]
When using Kustomize with overlays for each deployment, I do not want to be required to pass in namePrefix. We deploy to separate clusters per environment, so I would like the name to stay as set in the base deployment template.

https://github.com/argoproj/argo-cd/blob/master/manifests/ha/install.yaml#L189

**Comment by alexec** [comment]
Should be fixed in v1.0.0

**Comment by denniswebb** [comment]
Currently I'm working around this issue by adding the following to my argocd-cm configmap
```yaml
  configManagementPlugins: |
    - name: kustomize-basic
      generate:
        command: [sh, -c]
        args: ["kustomize build"]
```

**Comment by alexmt** [comment]
@alexec can we close it?

**Comment by denniswebb** [comment]
Is there a PR to fix it yet? If not I'll volunteer to attempt the fix.

**Comment by alexec** [comment]
This should be fixed in v1.0.0-rc1, but I'd like @denniswebb to verify the fix works for him.

**Comment by alexec** [comment]
I'm wrong. It is not completely fixed. It's a case of re-creating the manifests (i.e. `make manifests`). 

@alexmt do we need to backport to 1.0?
